controllers/activemqartemis_controller_cert_manager_test.go

```diff
@@ -0,0 +1,316 @@
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+package controllers
+
+import (
+	"encoding/json"
+	"os"
+
+	brokerv1beta1 "github.com/artemiscloud/activemq-artemis-operator/api/v1beta1"
+	"github.com/artemiscloud/activemq-artemis-operator/pkg/utils/certutil"
+	"github.com/artemiscloud/activemq-artemis-operator/pkg/utils/common"
+	cmv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	brokerCrNameBase = "broker-cert-mgr"
+
+	rootIssuerName       = "root-issuer"
+	rootCertName         = "root-cert"
+	rootCertNamespce     = "cert-manager"
+	rootCertSecretName   = "artemis-root-cert-secret"
+	caIssuerName         = "broker-ca-issuer"
+	caPemTrustStoreName  = "ca-truststore.pem"
+	caTrustStorePassword = "changeit"
+)
+
+var (
+	serverCert   = "server-cert"
+	rootIssuer   = &cmv1.ClusterIssuer{}
+	rootCert     = &cmv1.Certificate{}
+	caIssuer     = &cmv1.ClusterIssuer{}
+	caBundleName = "ca-bundle"
+)
+
+type ConnectorConfig struct {
+	Name             string
+	FactoryClassName string
+	Params           map[string]string
+}
+
+var _ = Describe("artemis controller with cert manager test", Label("controller-cert-mgr-test"), func() {
+	var installedCertManager bool = false
+
+	BeforeEach(func() {
+		if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+			//if cert manager/trust manager is not installed, install it
+			if !CertManagerInstalled() {
+				Expect(InstallCertManager()).To(Succeed())
+				installedCertManager = true
+			}
+
+			rootIssuer = InstallClusteredIssuer(rootIssuerName, nil)
+
+			rootCert = InstallCert(rootCertName, rootCertNamespce, func(candidate *cmv1.Certificate) {
+				candidate.Spec.IsCA = true
+				candidate.Spec.CommonName = "artemis.root.ca"
+				candidate.Spec.SecretName = rootCertSecretName
+				candidate.Spec.IssuerRef = cmmetav1.ObjectReference{
+					Name: rootIssuer.Name,
+					Kind: "ClusterIssuer",
+				}
+			})
+
+			caIssuer = InstallClusteredIssuer(caIssuerName, func(candidate *cmv1.ClusterIssuer) {
+				candidate.Spec.SelfSigned = nil
+				candidate.Spec.CA = &cmv1.CAIssuer{
+					SecretName: rootCertSecretName,
+				}
+			})
+			InstallCaBundle(caBundleName, rootCertSecretName, caPemTrustStoreName)
+		}
+	})
+
+	AfterEach(func() {
+		if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+			UnInstallCaBundle(caBundleName)
+			UninstallClusteredIssuer(caIssuerName)
+			UninstallCert(rootCert.Name, rootCert.Namespace)
+			UninstallClusteredIssuer(rootIssuerName)
+
+			if installedCertManager {
+				Expect(UninstallCertManager()).To(Succeed())
+				installedCertManager = false
+			}
+		}
+	})
+
+	Describe("cert manager integration test", func() {
```

Comment:
`Describe("cert manager integration test"` is nested in  Describe("artemis controller with cert manager test", Label("controller-cert-mgr-test")`, is that required?

Reply:
done.